src/hddsupport.c

```
@@ -215,78 +188,104 @@ static int hddCreateOPLPartition(const char *name)
-        // Leave the attempt retryable without powering DEV9 down. sysShutdownDev9's DDIOC_OFF is a
-        // terminal hardware stop, not a reversible reference release within the current IOP session.
-        hddModulesLoading = 0;
-        LOG("HDD: No HardDisk Drive detected.\n");
-        setErrorMessageWithCode(_STR_HDD_NOT_CONNECTED_ERROR, ERROR_HDD_IF_NOT_DETECTED);
-        return HDD_LOADMODULES_STATUS_ERROR;
+    LOG("HDDSUPPORT LoadModules done\n");
+    return retStatus;
+}
+
+// Returns 1 for MBR/GPT, 0 for APA, and -1 if an error occured
+int hddDetectNonSonyFileSystem()
```

Cppcheck (2.21.0)

[style] The function 'hddDetectNonSonyFileSystem' should have static linkage since it is not used outside of its translation unit.

(staticFunction)